sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -579,41 +690,83 @@ public Read withTemplateCompatibility() {
         BATCH
     }
 
+    /**
+     * An enumeration type for the method to be used when reading data from BigQuery.
+     */
+    public enum Method {
+
+      /**
+       * The default behavior if no method is explicitly set. Currently {@link #GCS_EXPORT}.
+       */
+      DEFAULT,
+
+      /**
+       * Export data to Google Cloud Storage in Avro format and read data from that location. This
```

Comment:
Hm maybe then the functions should be called `readViaGcsExport` and `readViaParallelRead`? How do you see a user deciding which method to use: as a decision between Avro and RowProto, or as a decision between GCS export and Parallel Read?

Reply:
Eventually I believe that we may support multiple formats in the parallel read API -- for example, our own row proto plus Avro or possibly Apache Arrow. The user may eventually need to choose both the data access method (GCS or BQ parallel read API) and the format.

In the case of the function names, it seems like what's being validated is that the parse function is appropriately typed (e.g. it operates over SchemaAndRecord or SchemaAndRowProto), so it makes sense to me to keep Avro and RowProto rather than switch to e.g. readViaGcsExport and readViaParallelRead.